neo/rawio/blackrockrawio.py

```diff
@@ -1113,14 +1142,18 @@ def __get_event_segment_ids(self, raw_event_data, masks, nev_data_masks):
             # No pause or reset mechanism present for file version 2.1 and 2.2
             return np.zeros(len(raw_event_data), dtype=int)
 
-        elif self.__nev_spec == '2.3':
+        elif self.__nev_spec in ['2.3', '3.0']:
             reset_ev_mask = self.__get_reset_event_mask(raw_event_data, masks, nev_data_masks)
             reset_ev_ids = np.where(reset_ev_mask)[0]
 
             # consistency check for monotone increasing time stamps
-            # explicitly converting to int to allow for negative diff values
-            jump_ids = \
-                np.where(np.diff(np.asarray(raw_event_data['timestamp'], dtype=int)) < 0)[0] + 1
+            # - Use logical comparator (instead of np.diff) to avoid unsigned dtype issues.
+            # - Only consider handled/known event types.
+            mask_handled = np.any(np.vstack([value[nev_data_masks[key]] for key, value in masks.items()]), axis=0)
```

Comment:
Are you sure the `vstack` is needed here? I think you don't need it, unless you noticed a performance improvement when using it.